### PR TITLE
[BUGFIX] fix model zoo parallel download

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
Fixes https://github.com/apache/incubator-mxnet/issues/17332 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/gluon/model_zoo/model_store.py [hunk 1]
```diff
@@ -103,16 +106,16 @@ def get_model_file(name, root=os.path.join(base.data_dir(), 'models')):
 
     util.makedirs(root)
 
-    zip_file_path = os.path.join(root, file_name+'.zip')
+    temp_zip_file_path = os.path.join(root, file_name+random_uuid+'.zip')
```
Comment (apeforest) [inline python/mxnet/gluon/model_zoo/model_store.py]: Do you want to use tempfile so it takes care of clean up automatically: https://docs.python.org/3/library/tempfile.html

File: tests/python/unittest/test_gluon_model_zoo.py [hunk 2]
```diff
@@ -19,12 +19,28 @@
...
+@with_seed()
+def test_parallel_download():
+    def fn(model_name):
+        model = get_model(model_name, pretrained=True)#, root='parallel_model/')
+        print(type(model))
+
+    threads = []
+    name = 'mobilenetv2_0.25'
+    for _ in range(10):
+        x = threading.Thread(target=fn, args=(name,))
```
Comment (apeforest) [inline tests/python/unittest/test_gluon_model_zoo.py]: Why not use multiprocess to test? My understand is that in horovod get_model is parallelized at process level not thread level?

File: python/mxnet/gluon/model_zoo/model_store.py (hunk 1 above)
Comment (leezu) [reply python/mxnet/gluon/model_zoo/model_store.py]: I believe tempfile is typically on a different filesystem, so there is no atomic rename operation to the target directory and filename available.

Comment (apeforest) [reply python/mxnet/gluon/model_zoo/model_store.py]: @leezu What do you mean by a different filesystem? Regarding name, I think you can also specify a file name using the NamedTemporaryfile class?

Comment (leezu) [reply python/mxnet/gluon/model_zoo/model_store.py]: You're right, I overlooked the `dir` argument.

**Comment by eric-haibin-lin** [comment]
Moved to https://github.com/apache/incubator-mxnet/pull/17372

**Pull request closed without merging**